ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,1421 @@
+
+Notes:
+* if the service task has an input variable mapping then it produces one or more `Variable:Created` events for the new local variables
+* if the job is completed with variables then these variables will be stored in the state until they are used for the output mapping (i.e. temporal/transient variables). They must therefore also be written into the `Job:Completed` event.
+* if the service task has an output variable mapping, or the job is completed with variables, then it produces one or more `Variable:Created`/`Variable:Updated` events for the new and updated variables of the workflow instance or scope
+* the order of writing follow-up events matters. Both `ServiceTask:Element_Activated` and `Variable:Created` need to be replayed before `Job:Created`.
+
+On processing:
+* the `ServiceTask:Activate_Element` command transitions the service task
+  element to activated, may create variables for input mapping and creates a job
+* the `JobBatch:Activate` command activates a batch of jobs
+* the `Job:Complete` command completes a job
+* the `ServiceTask:Complete_Element` command transitions the service task
+  element to completed, may create and/or update variables for output mapping
+  and marks the sequence flow as taken
+
+On replay:
+* the `ServiceTask:Element_Activating` event transitions the service task element to activating
+* the `Variable:Created` events create a variables for input mapping
+* the `ServiceTask:Element_Activated` event transitions the service task element to activated
+* the `Job:Created` event creates a job
+* the `JobBatch:Activated` event activates a batch of jobs
+* the `Job:Completed` event completes a job
+* the `Variable:Created` events create variables for output mapping
+* the `Variable:Updated` events update variables for output mapping
+* the `ServiceTask:Element_Completing` event transitions the service task element to completing
+* the `ServiceTask:Element_Completed` event transitions the service task element to completed
+* the `Sequence_Flow_Taken` event marks the sequence flow as taken
+* processing can continue again at `EndEvent:Activate_Element`
+
+Differences to current processing:
+* the event `Job:Created` is produced by the command `ServiceTask:Activate_Element` instead of a `Job:Create` command because it can't be interrupted
+
+Considered alternatives:
+* just like call activity variable propagation we could consider storing job
+  variables in `TemporaryVariable:Created` events
+
+### Message Intermediate Catch Event
+
+![message-correlation](ZEP-wf-stream-processing/event-sourcing-message-correlation-1.png)
+
+![message-correlation](ZEP-wf-stream-processing/event-sourcing-message-correlation-2.png)
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+IntermediateCatchEvent:Activate_Element | IntermediateCatchEvent:Element_Activating, IntermediateCatchEvent:Element_Activated, WorkflowInstanceSubscription:Opening, MessageSubscription:Open
```

Comment:
Interesting point. I'd like more opinions on this, because changing is a breaking change. We have the ability to break it now. They are also records that are not used by Operate/Tasklist, which makes this a less impactful breaking change. I just don't want to make this call alone. @npepinpe @saig0 WDYT?

Reply:
I don't see the cognitive friction with using `Open`, or at least it never confused me, but I'm also 100% aware that's purely anecdotal evidence. I'm alright with either, however as usual I'm for consistency, so if we change to `Create` we should do the same where we use `Open` elsewhere, and we should change `Close`/`Closed` to `Delete`/`Deleted`.

But as I said, I have no strong opinion here, so let's see if @saig0 does.

Reply:
Good point. I like that the names are consistent :+1: 

But I think it is not important for the ZEP. I would see it as an optional change. Depending on the load and the backward-compatibility requirements, I would like to change the names.